tough/src/schema/decoded.rs

```diff
@@ -126,6 +126,36 @@ impl Encode for EcdsaPem {
     }
 }
 
+/// [`Decode`]/[`Encode`] implementation for ECDSA public keys.
+/// This is a flexible implementation, it will try to decode the key assuming
+/// it is PEM encoded, if the decode fails it will then try to decode it
+/// assuming it's Hex encoded.
+/// The official TUF specification suggests ECDSA keys to be PEM encoded,
+/// however the go-tuf implementation encodes them as Hex numbers.
+/// This flexible decoder tries to cover both cases in a transparent way.
+#[derive(Debug, Clone, Copy)]
+pub struct EcdsaFlex {}
+
+impl Decode for EcdsaFlex {
+    fn decode(s: &str) -> Result<Vec<u8>, Error> {
```

Comment:
Hmm, unfortunately the trait definition doesn't allow us to know which type of encoding was encountered. If I understand this correctly it means that if we deserialize a key with hex encoding we would later serialize it with PEM encoding.

Reply:
> Hmm, unfortunately the trait definition doesn't allow us to know which type of encoding was encountered. If I understand this correctly it means that if we deserialize a key with hex encoding we would later serialize it with PEM encoding.

I'm less concerned with this since @flavio doesn't need to write hex-encoded Ecds keys anyway. Right now we just need to support reading them. Also, I believe reserializing works fine because the `Decoded<T>` struct has the original string in it.